component/config.go

```diff
@@ -110,6 +112,46 @@ func callValidateIfPossible(v reflect.Value) error {
 // Type is the component type as it is used in the config.
 type Type string
 
+var _ fmt.Stringer = (Type)("")
+
+// String returns the string representation of the type.
+func (t Type) String() string {
+	return string(t)
+}
+
+// typeRegexp is used to validate the type of a component.
+// A type must start with an ASCII alphabetic character and
+// can only contain ASCII alphanumeric characters and '_'.
+var typeRegexp = regexp.MustCompile(`^[a-zA-Z][0-9a-zA-Z_]*$`)
+
+// newType creates a type. It returns an error if the type is invalid.
+// A type must
+// - have at least one character,
+// - start with an ASCII alphabetic character and
+// - can only contain ASCII alphanumeric characters, '-' and '_'.
```

Comment:
This comment mentions `-` but I dont see `-` being allowed in `typeRegexp`

Reply:
Fixed!